bootstrap/controllers/ck8sconfig_controller.go

```diff
@@ -258,7 +258,13 @@ func (r *CK8sConfigReconciler) joinControlplane(ctx context.Context, scope *Scop
 		return err
 	}
 
-	snapInstallData := r.resolveInPlaceUpgradeRelease(machine)
+	snapInstallData := r.getSnapInstallDataFromSpec(scope.Config.Spec)
```

Comment:
Seems like `Config` is a pointer and can be nil, maybe we can check for that to prevent panics? I applicable, please have look at other occurrences as well.

Reply:
Eh, we create a Config struct and assign it at the start of the Reconcile function, so it's never nil. Also this would require change all over the codebase so it's out of scope for this PR.